elasticapm/instrumentation/packages/asyncio/httpcore.py

```diff
@@ -0,0 +1,120 @@
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+#  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+#  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+#  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+#  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+#  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+#  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from elasticapm.conf import constants
+from elasticapm.contrib.asyncio.traces import async_capture_span
+from elasticapm.instrumentation.packages.asyncio.base import AsyncAbstractInstrumentedModule
+from elasticapm.traces import DroppedSpan, execution_context
+from elasticapm.utils import default_ports, url_to_destination
+from elasticapm.utils.disttracing import TracingOptions
+
+
+class HTTPCoreAsyncInstrumentation(AsyncAbstractInstrumentedModule):
+    name = "httpcore"
+
+    instrument_list = [
+        ("httpcore._async.connection", "AsyncHTTPConnection.request"),
+    ]
+
+    async def call(self, module, method, wrapped, instance, args, kwargs):
+        if "method" in kwargs:
+            method = kwargs["method"].decode("utf-8")
+        else:
+            method = args[0].decode("utf-8")
+
+        # URL is a tuple of (scheme, host, port, path), we want path
+        if "url" in kwargs:
+            url = kwargs["url"][3].decode("utf-8")
+        else:
+            url = args[1][3].decode("utf-8")
+
+        headers = None
+        if "headers" in kwargs:
+            headers = kwargs["headers"]
+            if headers is None:
+                headers = []
+                kwargs["headers"] = headers
+
+        scheme, host, port = instance.origin
+        scheme = scheme.decode("utf-8")
+        host = host.decode("utf-8")
+
+        if port != default_ports.get(scheme):
+            host += ":" + str(port)
+
+        signature = "%s %s" % (method.upper(), host)
+
+        url = "%s://%s%s" % (scheme, host, url)
+        destination = url_to_destination(url)
+
+        transaction = execution_context.get_transaction()
+
+        async with async_capture_span(
+            signature,
+            span_type="external",
+            span_subtype="http",
+            extra={"http": {"url": url}, "destination": destination},
+            leaf=True,
+        ) as span:
+            # if httpcore has been called in a leaf span, this span might be a DroppedSpan.
+            leaf_span = span
+            while isinstance(leaf_span, DroppedSpan):
+                leaf_span = leaf_span.parent
+
+            if headers is not None:
+                # It's possible that there are only dropped spans, e.g. if we started dropping spans.
```

Comment:
This comment seems confusing, maybe "e.g. if we reached the transaction_max_spans limit"?

(I realize the irony that you copied this comment verbatim from the urllib3 instrumentation and that I wrote it originally :D )